components/zcash_note_encryption/src/lib.rs

```diff
@@ -462,6 +519,31 @@ impl<D: Domain> NoteEncryption<D> {
     }
 }
 
+#[cfg(feature = "encrypt-to-recipient")]
+impl<D: PayloadEncryptionDomain> NoteEncryption<D> {
+    /// Encrypts a payload that may decrypted with the same shared secret and epk as can
+    /// be used to decrypt the note plaintext for a given note. Returns a newly generated
+    /// nonce that must be used in decryption along with the ciphertext.
+    ///
+    /// This can be used to make a separate payload passed out-of-band decryptable to the
+    /// recipient of a Zcash payment.
+    pub fn encrypt_to_note_recipient<R: RngCore + CryptoRng>(
+        &self,
+        rng: &mut R,
+        kdf_personalization: &D::KdfPersonalization,
```

Comment:
The KDF personalization used for this feature should not be arbitrary, because then it can coincide with "Zcash_OrchardKDF" or some other already-used BLAKE2b-256 personalization, complicating security analysis. Also, it should be different depending on whether the domain is for Sapling or Orchard or some future shielded protocol. One option is to require it to start with, say, "ZcashSAK" or "ZcashOAK" (Zcash Sapling/Orchard Associated Key), with the remaining 8 bytes left arbitrary.

Reply:
I think that the name should remain `KdfPersonalization`, so that in the event that we move away from Blake2b in a future protocol and use a hash that requires a different approach to personalization, that the trait remains generic. But I will change the type at the implementation site and document it in those places as 8-byte suffixes.